File: problems/secondLargestNumberInArray/go/code.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
)

func main() {
	array := []int{99, 82, -34, 3, 15, 76, 102, 14, 82, 82, 99}

	// Argument with index zero is always program name
	args := os.Args[1:]

	if len(args) < 2 {
		fmt.Println("Number of input arguments is less then 2! Default array is used:")
		fmt.Println(array)
	} else {
	    fmt.Println(args)

		var array2 []int

		for i:=0; i<len(args); i++ {
			num, err := strconv.Atoi(args[i])
			if err != nil {
				panic(err)
			}
			array2 = append(array2, num)

		}

		array = array2
	}


	findMaximums(array)
}

func findMaximums(array []int) {
	firstMaximum := math.MinInt
	secondMaximum := math.MinInt

	for i:=0; i<len(array); i++ {
		if array[i]> firstMaximum {
			secondMaximum = firstMaximum
			firstMaximum = array[i]
		} else if array[i]<firstMaximum && array[i]> secondMaximum {
			secondMaximum = array[i]
		}
	}

	if (secondMaximum == math.MinInt32) {
        println("There is no second largest number in array. \n" +
                "Or array contains minimal integer value: -9223372036854775808. \n" +
                "Or it contains only duplicates. \n" +
                "Please try again with another input!")
        return
    }

	fmt.Printf("First maximum is: %v\n", firstMaximum)
	fmt.Printf("Second maximum is: %v\n", secondMaximum)
}

File: problems/secondLargestNumberInArray/typescript/code.ts
// The first two elements are 'node' and 'yourScript.ts', so we slice them off
const args: string[] = process.argv.slice(2);
let array: Array<number> = []

if (args.length < 2) {
    console.log("Number of input arguments is less then 2! Default array is used:")
    array = [18, 2, 3, 3, 44, 0, 77, -99, -42, 82]
    console.log(array)
} else {
    for (let argument of args) {
        array.push(Number(argument))
    }
    console.log(array)
}

findMaximums(array)


function findMaximums(array: number[]) {
    let firstMaximum = Number.NEGATIVE_INFINITY
    let secondMaximum = Number.NEGATIVE_INFINITY

    for (let i: number = 0; i < array.length; i++) {
        let value = array[i]

        if (value > firstMaximum) {
            secondMaximum = firstMaximum
            firstMaximum = value
        } else if (value < firstMaximum && value > secondMaximum) {
            secondMaximum = value
        }
    }

    if (secondMaximum === Number.NEGATIVE_INFINITY) {
        console.log("There is no second largest number in array.\n" +
            `Or array contains minimal numeric value: ${Number.NEGATIVE_INFINITY}.\n` +
            `Or it contains only duplicates.\n` +
            `Please try again with another input!\n`)

        return
    }

    console.log(`First maximum is: ${firstMaximum}`)
    console.log(`Second maximum is: ${secondMaximum}`)
}
